caluma/caluma_form/structure.py

```diff
@@ -0,0 +1,218 @@
+"""Hierarchical representation of a document / form."""
+import weakref
+from functools import singledispatch
+from typing import Optional
+
+from .models import Question
+
+
+def object_local_memoise(method):
+    def new_method(self, *args, **kwargs):
+        if not hasattr(self, "_memoise"):
+            self._memoise = {}
+
+        key = str([args, kwargs, method])
+        if key in self._memoise:
+            return self._memoise[key]
+        ret = method(self, *args, **kwargs)
+        self._memoise[key] = ret
+        return ret
+
+    return new_method
+
+
+class Element:
+    def __init__(self, parent=None):
+        self._parent = weakref.ref(parent) if parent else None
+
+    def parent(self):
+        return self._parent() if self._parent else None
+
+    def children(self):  # pragma: no cover
```

Comment:
`get_children`

Reply:
I'm not a fan of naming things `get_XXX`. It's kind of java-think IMO

Reply:
Then make it a property - this doesn't even do anything or am I wrong?

Reply:
This is overloaded in the subclasses where the implementation actually needs to do things

Reply:
Ok